qiime2/sdk/plugin_manager.py

```diff
@@ -38,29 +38,30 @@ def iter_entry_points(cls):
 
     # This class is a singleton as it is slow to create, represents the
     # state of a qiime2 installation, and is needed *everywhere*
-    def __new__(cls):
+    def __new__(cls, install_plugins=True):
         if cls.__instance is None:
             self = super().__new__(cls)
-            self._init()
+            self._init(install_plugins=install_plugins)
             cls.__instance = self
         return cls.__instance
 
-    def _init(self):
+    def _init(self, install_plugins):
         self.plugins = {}
         self.semantic_types = {}
         self.transformers = collections.defaultdict(dict)
         self.formats = {}
         self.views = {}
         self.type_formats = []
 
-        # These are all dependent loops, each requires the loop above it to
-        # be completed.
-        for entry_point in self.iter_entry_points():
-            plugin = entry_point.load()
-            self.plugins[plugin.name] = plugin
+        if install_plugins:
+            # These are all dependent loops, each requires the loop above it to
+            # be completed.
+            for entry_point in self.iter_entry_points():
+                plugin = entry_point.load()
+                self.plugins[plugin.name] = plugin
 
-        for plugin in self.plugins.values():
-            self._integrate_plugin(plugin)
+            for plugin in self.plugins.values():
+                self._integrate_plugin(plugin)
```

Comment:
So it has occurred to me that this loop might be happening after setting up the plugin dict on purpose, which `install_plugin` can't do without a `finalize()` method of some kind.